core/src/main/java/org/apache/carbondata/core/cache/CacheProvider.java

```diff
@@ -148,7 +177,11 @@ private boolean dictionaryCacheAlreadyExists(CacheType cacheType) {
    * Below method will be used to clear the cache
    */
   public void dropAllCache() {
-    cacheTypeToLRUCacheMap.clear();
-    cacheTypeToCacheMap.clear();
+    if(null != carbonLRUCache) {
+      carbonLRUCache.clear();
+    }
+    if(null != cacheTypeToCacheMap) {
```

Comment:
null check is not required for cacheTypeToCacheMap as it gets initialized when the class is first refrenced